doc/source/data/api/dataset.rst

```diff
@@ -0,0 +1,276 @@
+.. _dataset-api:
+
+Dataset API
+===========
+
+.. autoclass:: ray.data.Dataset
+
+.. tabbed:: Basic Transformations
+
+    .. autosummary::
+
+        ray.data.Dataset.map
+        ray.data.Dataset.map_batches
+        ray.data.Dataset.flat_map
+        ray.data.Dataset.filter
+        ray.data.Dataset.add_column
+        ray.data.Dataset.drop_columns
+        ray.data.Dataset.random_sample
+        ray.data.Dataset.limit
+
+.. tabbed:: Sorting, Shuffling, Repartitioning
+
+    .. autosummary::
+
+        ray.data.Dataset.sort
+        ray.data.Dataset.random_shuffle
+        ray.data.Dataset.randomize_block_order
+        ray.data.Dataset.repartition
+
+.. tabbed:: Splitting and Merging Datasets
```

Comment:
May be a bit tricky going forward, merging ops are transformations and splitting ops are consumptions.

Reply:
Any concrete suggestions for how to structure this in this PR? (As a user I don't think I'd immediately go to `Consuming Datasets` if I were looking to split my dataset.